src/containers/Profile/Profile.tsx

```diff
@@ -1,42 +1,47 @@
-import * as React from 'react'
+import React, { useCallback, useEffect, useMemo, useRef } from 'react'
+import { useDispatch, useSelector } from 'react-redux'
 import { Profile as BaseProfile } from 'decentraland-ui/dist/components/Profile/Profile'
+import { loadProfileRequest } from '../../modules/profile/actions'
+import { getData as getProfiles } from '../../modules/profile/selectors'
 import { Props } from './Profile.types'
 
-export default class Profile extends React.PureComponent<Props> {
-  static defaultProps = {
-    inline: true
-  }
+const Profile = function<T extends React.ElementType>(props: Props<T>) {
+  const { address, debounce, inline = true } = props
+  const profiles = useSelector(getProfiles)
+  const dispatch = useDispatch()
 
-  timeout: NodeJS.Timeout | null = null
+  const timeoutRef = useRef<NodeJS.Timeout>()
 
-  componentWillMount() {
-    this.fetchProfile(this.props)
-  }
+  const avatar = useMemo(() => {
+    const profile = profiles[address]
+    return profile ? profile.avatars[0] : null
+  }, [address, profiles])
 
-  componentDidUpdate(prevProps: Props) {
-    if (prevProps.address !== this.props.address) {
-      this.fetchProfile(this.props)
-    }
-  }
+  const onLoadProfile: typeof loadProfileRequest = useCallback(
+    (address: string) => dispatch(loadProfileRequest(address)),
+    [address]
+  )
 
-  fetchProfile(props: Props) {
-    const { address, avatar, debounce, onLoadProfile } = props
+  useEffect(() => {
     if (!avatar) {
       if (debounce) {
-        if (this.timeout) {
-          clearTimeout(this.timeout)
+        if (timeoutRef.current) {
+          clearTimeout(timeoutRef.current)
         }
-        this.timeout = setTimeout(() => {
-          onLoadProfile(address)
-          this.timeout = null
-        }, debounce)
+        timeoutRef.current = setTimeout(() => onLoadProfile(address), debounce)
       } else {
         onLoadProfile(address)
       }
     }
-  }
 
-  render() {
-    return <BaseProfile {...this.props} />
-  }
+    return () => {
+      if (timeoutRef.current) {
+        clearTimeout(timeoutRef.current)
+      }
+    }
+  }, [address, timeoutRef])
```

Comment:
If I'm not wrong, only address needs to be taken into account to trigger the `useEffect` here. If not, the effect will be re-triggered multiple times as the `timeoutRef` will change every time the `useEffect` is executed and there's no avatar and there's a debounce.
```suggestion
  }, [address])
```